i2c/mcp9808_i2c/mcp9808_i2c.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/i2c.h"
+
+/* Example code to talk to a MCP9808 ±0.5°C Digital temperature Sensor
+   
+   This reads and writes to registers on the board. 
+
+   Connections on Raspberry Pi Pico board, other boards may vary.
+
+   GPIO PICO_DEFAULT_I2C_SDA_PIN (On Pico this is 4 (pin 6)) -> SDA on MCP9808 board
+   GPIO PICO_DEFAULT_I2C_SCK_PIN (On Pico this is 5 (pin 7)) -> SCL on MCP9808 board
+   Vsys (pin 39) -> VDD on MCP9808 board
+   GND (pin 38)  -> GND on MCP9808 board
+
+*/
+//The bus address is determined by the state of pins A0, A1 and A2 on the MCP9808 board
+static uint8_t ADDRESS = 0X18;
+
+const uint8_t WRITE_MODE = 0XFE;
+const uint8_t READ_MODE = 0XFF;
+
+
+//hardware registers
+
+const uint8_t REG_POINTER = 0X00;
+const uint8_t REG_CONFIG = 0X01;
+const uint8_t REG_TEMP_UPPER = 0X02;
+const uint8_t REG_TEMP_LOWER = 0X03;
+const uint8_t REG_TEMP_CRIT = 0X04;
+const uint8_t REG_TEMP_AMB = 0X05;
+const uint8_t REG_MAN_ID = 0X06;
+const uint8_t REG_DEV_ID = 0X07;
+const uint8_t REG_RESOLUTION = 0X08;
+
+
+void mcp9808_check_limits(uint8_t upper_byte) {
+
+    // Check flags and raise alerts accordingly 
+    if ((upper_byte & 0x40) == 0x40){ //TA > TUPPER
+        printf("Temperature is above the upper temperature limit.\n");
+    }
+    if ((upper_byte & 0x20) == 0x20) { //TA < TLOWER
+        printf("Temperature is below the lower temperature limit.\n");
+    }
+    if ((upper_byte & 0x80) == 0x80) { //TA < TCRIT

```

Comment:
Should the comment here be `//TA > TCRIT` ? (which is what the printf is saying?)

Reply:
Whoops yes